frontend/public/components/hpa.tsx

```diff
@@ -31,8 +35,98 @@ const menuActions = [
   ...common,
 ];
 
-export const HorizontalPodAutoscalersDetails: React.SFC<HorizontalPodAutoscalersDetailsProps> = ({obj: hpa}) => {
-  return <div className="co-m-pane__body">
+const humanizeResourceValue = (value: string, type: string) => {
+  if (!value) {
+    return null;
+  }
+
+  switch (type) {
+    case 'cpu':
+      return humanizeCPU(value);
+    case 'memory':
+      return humanizeMem(value);
+    default:
+      return value;
+  }
+};
+
+const MetricsTable: React.SFC<MetricsTableProps> = ({obj: hpa}) => {
+  const namespace = hpa.metadata.namespace;
+  // https://github.com/kubernetes/api/blob/master/autoscaling/v2beta1/types.go
+  const metrics = hpa.spec.metrics.map((metric, i) => {
+    let typeLabel, targetValue, currentValue;
+    const currentMetrics = _.get(hpa, ['status', 'currentMetrics', i]);
+    switch (metric.type) {
+      case 'External':
+        typeLabel = metric.external.metricName;
+        // TODO: show metric selector for external metrics?
```

Comment:
I would suggest trying to avoid nested `if` statements within a `switch`.  It's hard to follow the flow & pretty easy to break in later updates.  The  valueless`let typeLabel, targetValue, currentValue;` at the top is a pretty good indicator we can break it down into smaller bits.

Reply:
See my previous comment, I'd even wager you can avoid the `switch` altogether to improve readability.